### PR TITLE
kafka: Add support for using TLS authentication for the kafka output

### PR DESCRIPTION
**Pull request by Ormod**
With the advent of Kafka 0.9.0+ it is possible to set up TLS client
certificate based authentication to limit access to Kafka.

Four new configuration variables are specified for setting up the
authentication. If they're not set the behavior stays the same as
before the change.

### CODE REVIEW

**Comment by sparrc** [comment]
Looks like you have a formatting error, try running `go fmt ./...`

**Comment by Ormod** [comment]
Fixed the format issue. The test fails seem to be related to some of the (non-kafka) plugin tests failing.

**Comment by Ormod** [comment]
Rebased the PR so it merges cleanly again.

**Comment by Ormod** [comment]
Anything I can do to hurry this along?

File: plugins/outputs/kafka/kafka.go [hunk 1]
```diff
@@ -28,10 +40,52 @@ var sampleConfig = `
...
+func createTlsConfiguration(k *Kafka) (t *tls.Config) {
+	if k.Certificate != "" && k.Key != "" && k.CA != "" {
+		cert, err := tls.LoadX509KeyPair(k.Certificate, k.Key)
+		if err != nil {
+			log.Fatal(err)
+		}
+
+		caCert, err := ioutil.ReadFile(k.CA)
+		if err != nil {
+			log.Fatal(err)
```
Comment (sparrc) [inline plugins/outputs/kafka/kafka.go]: `log.Fatal` calls `os.Exit(1)`, which shouldn't happen in here. This function should return an error which should be returned by the `Connect()` function.

**Comment by sparrc** [comment]
Sorry for the delay @Ormod, just address the `log.Fatal` call and I can merge

**Comment by Ormod** [comment]
Thanks for the review, I'll try to get this sorted tomorrow once next to a computer again.

**Pull request closed without merging**